theano/sandbox/gpuarray/blocksparse.py

```diff
@@ -0,0 +1,142 @@
+from __future__ import absolute_import, print_function, division
+import logging
+import os
+
+import numpy
+from theano import Apply, tensor
+from theano.gof import COp
+from theano.tensor import discrete_dtypes
+
+from theano.gradient import grad_undefined
+
+from .type import gpu_context_type
+from .basic_ops import as_gpuarray_variable, infer_context_name
+
+_logger = logging.getLogger('theano.sandbox.gpuarray.blocksparse')
+
+
+class GpuSparseBlockGemv(COp):
+    """
+    GPU version of SparseBlockGemv. Check SparseBlockGemv's docstring for more
+    information.
+
+    This should not be directly called since the interface is subject
+    to change without notice.  Use the sandbox.blocksparse.sparse_block_dot()
+    function for a stable interface.
+    """
+    __props__ = ('inplace',)
+    params_type = gpu_context_type
+
+    def __init__(self, inplace=False):
+        COp.__init__(self, "blockgemv.c", "APPLY_SPECIFIC(blockgemv)")
+        self.inplace = inplace
+        if self.inplace:
+            self.destroy_map = {0: [0]}
+
+    def get_params(self, node):
+        return node.inputs[0].type.context
+
+    def get_op_params(self):
+        if self.inplace:
+            return [('INPLACE', '1')]
+        else:
+            return []
+
+    def c_header_dirs(self):
+        return [os.path.dirname(__file__)]
+
+    def c_headers(self):
+        return ['<gpuarray/buffer_blas.h>', '<gpuarray/buffer.h>',
+                '<gpuarray_helper.h>']
+
+    def make_node(self, o, W, h, inputIdx, outputIdx):
+        ctx = infer_context_name(o, W, h)
+        o = as_gpuarray_variable(o, ctx)
+        W = as_gpuarray_variable(W, ctx)
+        h = as_gpuarray_variable(h, ctx)
+        assert o.ndim == 3
```

Comment:
I mean, in the futur, maybe we want to allow this to be on the GPU, but for now, we don't want to go there.